src/Symfony/Component/Notifier/Bridge/AmazonSns/Tests/AmazonSnsTransportTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\AmazonSns\Tests;
+
+use AsyncAws\Sns\Result\PublishResponse;
+use AsyncAws\Sns\SnsClient;
+use Symfony\Component\Notifier\Bridge\AmazonSns\AmazonSnsOptions;
+use Symfony\Component\Notifier\Bridge\AmazonSns\AmazonSnsTransport;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Tests\TransportTestCase;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class AmazonSnsTransportTest extends TransportTestCase
+{
+    public function createTransport(?HttpClientInterface $client = null): TransportInterface
+    {
+        return (new AmazonSnsTransport(new SnsClient(['region' => 'eu-west-3']), $client ?: $this->createMock(HttpClientInterface::class)))->setHost('host.test');
+    }
+
+    public function toStringProvider(): iterable
+    {
+        yield ['sns://host.test?region=eu-west-3', $this->createTransport()];
+    }
+
+    public function supportedMessagesProvider(): iterable
+    {
+        yield [new SmsMessage('0601020304', 'Hello!')];
+        yield [new ChatMessage('Hello', new AmazonSnsOptions())];
```

Comment:
```suggestion
        yield [new ChatMessage('Hello')];
        yield [new ChatMessage('Hello', new AmazonSnsOptions())];
```

this should also be supported, right?

Reply:
No, because recipient (in AmazonSnsOptions) is required to send a correct chat message. I can change support in transport class but it will always return an HTTP error from amazon SNS. Same if you don't add a recipient in AmazonSnsOptions but it's more easy to understand why.